tests/rest/client/test_relations.py

```diff
@@ -241,6 +241,60 @@ def test_repeated_paginate_relations(self):
         found_event_ids.reverse()
         self.assertEquals(found_event_ids, expected_event_ids)
 
+    def test_pagination_from_sync_messages(self):
```

Comment:
I'm having trouble following this one in general; anything you can do to clarify would be a big win I think, however I will try and make some notes about what I'm struggling with..